### PR TITLE
Unit tests

### LINKED ISSUE #3: Improve unit test code coverage
Improve code coverage to > 80%.

### PR DESCRIPTION
**Pull request by rockspore**
Resolves #3 

The coverage will certainly change as the cli evolves. At least for now it is at 80%, ;-)

### CODE REVIEW

File: cmd/provision/legacy.go [hunk 1]
```diff
@@ -201,13 +201,13 @@ func (p *provision) createLegacyCredential(printf shared.FormatFn) (*keySecret,
...
 
-	resp, err := p.ApigeeClient.Do(req, nil)
+	_, err = p.ApigeeClient.Do(req, nil)
 	if err != nil {
 		return nil, err
 	}
-	if resp.StatusCode > 299 {
-		return nil, fmt.Errorf("creating credential, status: %d", resp.StatusCode)
-	}
+	// if resp.StatusCode > 299 {
```
Comment (theganyo) [inline cmd/provision/legacy.go:208]: If we don’t need this, we should remove it.

File: cmd/provision/legacy.go [hunk 2]
```diff
@@ -233,13 +233,13 @@ func (p *provision) verifyInternalProxy(client *http.Client, printf shared.Forma
...
 	}
-	if err != nil {
+	if err == nil {
 		res, err = client.Do(req)
 		if res != nil {
 			defer res.Body.Close()
 		}
 	}
-	if err != nil {
+	if res.StatusCode > 299 || err != nil {
```
Comment (theganyo) [inline cmd/provision/legacy.go:242]: Do we know for certain that res will not be nil?

File: cmd/provision/legacy.go (hunk 1 above)
Comment (rockspore) [reply cmd/provision/legacy.go:208]: Sure I will remove it. ApigeeClient explicitly checks the status code and return error on non-2XX codes.

File: cmd/provision/legacy.go (hunk 2 above)
Comment (rockspore) [reply cmd/provision/legacy.go:242]: Good point. Will add the null check.

**Comment by theganyo** [comment]
Thanks!

**Pull request merged**